src/domain/balances/balances.repository.ts

```diff
@@ -32,10 +19,9 @@ export class BalancesRepository implements IBalancesRepository {
     trusted?: boolean;
     excludeSpam?: boolean;
   }): Promise<Balance[]> {
-    // TODO: route TransactionApi balances retrieval from BalancesApiManager
-    return this.balancesApiManager.useExternalApi(args.chainId)
-      ? this._getBalancesFromBalancesApi(args)
-      : this._getBalancesFromTransactionApi(args);
+    const api = await this.balancesApiManager.getBalancesApi(args.chainId);
```

Comment:
This is the main point of the refactor: this repository asks for an `api` to the manager and then invokes the `getBalances` function, without knowing any internal detail of the API being used, or the mappings needed. All the logic is encapsulated in the underlying `api`.